packages/expect/src/jest-utils.ts

```diff
@@ -334,7 +334,7 @@ export function iterableEquality(a: any, b: any, customTesters: Array<Tester> =
     return iterableEquality(
       a,
       b,
-      [...filteredCustomTesters],
+      [...customTesters],
```

Comment:
I'm not sure if this fix is entirely fool-proof for more complicated case, but at least this recursion with `filteredCustomTesters` seems redundant.

Previously `iterableEqualityWithStack` was calling `iterableEquality(..., [... iterableEqualityWithStack], ...)`, but `iterableEquality` is already setting up `filteredCustomTesters`.